groupsforsections.py

```diff
@@ -0,0 +1,85 @@
+import json
+import sys
+import logging
+import requests
+import urllib
+import utils
+import yaml
+
+
+def create_group_category():
+    logging.debug('create_group_category() Called')
+
+    url = canvas_url + '/api/v1/courses/' + course_id + '/group_categories?name=' + urllib.quote_plus(
+            group_category_name)
+    response = api_handler(url, 'Post')
+    if response is None:
+        logging.error('Api call for \'creating group categories\' is not successful')
+        sys.exit(1)
+
+    if response.status_code != requests.codes.ok:
+        error_msg_handler(response)
+        sys.exit(1)
+    group_category = json.loads(response.text)
+    return group_category['id']
+
+
+def api_handler(url, request_type):
+    logging.debug('api_handler() called')
+    logging.info('URL: ' + url)
+    response = None
+    headers = {'content-type': 'application/json', 'Authorization': 'Bearer' + ' ' + canvas_token}
+    try:
+        if request_type == 'Get':
+            response = requests.get(url, headers=headers)
+        elif request_type == 'Post':
+            response = requests.post(url, headers=headers)
+    except requests.exceptions.RequestException:
+        logging.exception('api_handler() has some errors')
+
+    return response
+
+
+def error_msg_handler(response):
+    # using yaml instead of json.load to get the error message as logs  will show non unicode values in logs
+    # for eg., [{u'message': u'Invalid access token.'}] ---> [{'message': 'Invalid access token.'}]
+
+    error_res = yaml.safe_load(response.text)
+    logging.error('Api response has some errors: ' + str(error_res['errors']) + ' and Response Code: ' + str(
+            response.status_code))
+
+utils.setup_logging()
+logging.info('Script Started')
+
+logging.debug('args: ' + str(sys.argv))
+if len(sys.argv) is not 2:
+    logging.error("The command line argument (path to properties file \'config.yaml\') is not provided")
+    sys.exit(1)
+
+config_file = sys.argv[1]
+
+logging.debug("reading the \'config.yaml\'")
+
+with open(config_file, 'r') as yml_file:
+    cfg = yaml.load(yml_file)
+
```

Comment:
What happens below when accessing elements of `cfg` when the above initialization fails?  You should check for the presence of the keys before using them.

Also, `cfg` should be initialized to an empty dictionary before the `with` clause.  Or else test whether it's been defined after that clause.